### PR TITLE
DAOS-12418 utils: need to make sure ca is not NULL

### PR DESCRIPTION
**Pull request by wiliamhuang**
Have to make sure ca is not NULL before freeing
ca->src and ca->dst in cont_clone_hdlr().

Required-githooks: true

Signed-off-by: Lei Huang <lei.huang@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/utils/daos_hdlr.c [hunk 1]
```diff
@@ -2818,8 +2818,10 @@ cont_clone_hdlr(struct cmd_args_s *ap)
 out:
 	D_FREE(src_str);
 	D_FREE(dst_str);
-	D_FREE(ca->src);
-	D_FREE(ca->dst);
+	if (ca) {
+		D_FREE(ca->src);
+		D_FREE(ca->dst);
+	}
```
Comment (daltonbohning) [inline src/utils/daos_hdlr.c:2824]: It looks like we also don't `D_FREE(ca)`? (alloc'ed at the beginning of this function).

**Review by daltonbohning — APPROVED** [review]

File: src/utils/daos_hdlr.c (hunk 1 above)
Comment (wiliamhuang) [reply src/utils/daos_hdlr.c:2824]: Looks like the memory of ca will be accessed later at,
https://github.com/daos-stack/daos/blob/master/src/control/cmd/daos/container.go#L905-L906
Maybe I can add code to free ap.dm_args (ca) in Go.

Comment (daltonbohning) [reply src/utils/daos_hdlr.c:2824]: I see... Anyway, not a blocker for this fix

Comment (wiliamhuang) [reply src/utils/daos_hdlr.c:2824]: It looks like the memory is freed in Go code already. No worry about freeing ca here.
https://github.com/daos-stack/daos/blob/master/src/control/cmd/daos/util.go#L193-L195
	if ap.dm_args != nil {
		C.free_daos_alloc(unsafe.Pointer(ap.dm_args))
	}

Comment (daltonbohning) [reply src/utils/daos_hdlr.c:2824]: Thanks for checking on that!

**Review by knard-intel — APPROVED** [review]
LGTM

**Review by mjmac — APPROVED** [review]

**Comment by mjmac** [comment]
@wiliamhuang: Thanks for this fix. I've incorporated it into the backport PR which hasn't landed yet.

**Pull request merged**